### ISSUE TITLE
can't open zip file  Unhandled exception: Unsupported zip archive [ZippyError]

### ISSUE THREAD

**Issue by bung87** [guzba/zippy#11]
https://github.com/bung87/finder/tree/zippy  

I previously use std zip lib which works fine and test passed, when switch to zippy I got ` Unhandled exception: Unsupported zip archive [ZippyError]`  

the zip file compression through mac context menu, am on macOS , nim 1.4.4

**Comment by guzba** [comment]
This happens when the general purpose value has the "data descriptor" bit set. I do not support this. I've not seen a zip archive with it used. How did you create a the zip archive with this flag set? I'd be curious to see if I can reproduce that.

https://users.cs.jmu.edu/buchhofp/forensics/formats/pkzip.html
https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

**Comment by bung87** [comment]
I checked the general purpose flag , `echo tobin(generalPurposeFlag.int,16)` on mac outputs `0000000000001000`
https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT described as 
Bit 4: Reserved for use with method 8, for enhanced
               deflating.

**Comment by guzba** [comment]
Hm, it looks like Mac is writing Zip64 files sometimes. I know I've tested with Mac archives that worked fine in the past.

Zip64 are different from standard Zip files even though they share an extension. Zip64 is generally supported in most places. There isn't much for docs on this and they are not clear as to what exactly 'enhanced deflating" is. I'm assuming it means Deflate64 instead of standard Deflate for compression.

Deflate64 is not the same as standard Deflate. It is different enough to require quite a bit of work and is not something I plan to work on in the short term, sorry.

**Comment by bung87** [comment]
After google found deflate64 also called "enhanced deflate" is PKWare proprietary, that's why I can't find any open source implementation.

**Comment by guzba** [comment]
It's interesting that Mac is generating zip archives with enhanced deflate. If we learn more we can add it to this issue. I'm going to add "deflate64" to the issue title too quick.

**Comment by bung87** [comment]
yeah , I think std zip module can work because it's zip header point to PKWare's header not the open source zlib
